x-pack/plugins/security_solution/public/detection_engine/rule_creation/logic/esql_validator.ts

```diff
@@ -34,11 +38,40 @@ const constructValidationError = (error: Error) => {
   };
 };
 
+const constructSyntaxError = (error: Error) => {
+  return {
+    code: ERROR_CODES.INVALID_SYNTAX,
+    message: error?.message
+      ? i18n.esqlValidationErrorMessage(error.message)
+      : i18n.ESQL_VALIDATION_UNKNOWN_ERROR,
+    error,
+  };
+};
+
 /**
  * checks whether query has metadata _id operator
  */
-export const computeHasMetadataOperator = (esqlQuery: string) => {
-  return /(?<!\|[\s\S.]*)\s*metadata[\s\S.]*_id[\s\S.]*/i.test(esqlQuery?.split('|')?.[0]);
+export const computeHasMetadataOperator = (ast: ESQLAst) => {
+  const commands = ast.filter((a) => a.type === 'command');
+  const fromCommand = commands.find((command) => command.name === 'from');
```

Comment:
can we juse use one `find` here?
```suggestion
  const fromCommand = ast.find((command) => a.type === 'command' && a.name === 'from');
```